### PR TITLE
Add Into<&str> for optional parameters

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Remains to call `.into()` on the arguments (didn't look yet where I can do it in the code).

cc @EPashkin

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Actually, I could just implement the implementation for `Into<Option<&'a str>>` instead. Much easier this way!

File: src/analysis/bounds.rs [hunk 1]
```diff
@@ -50,8 +55,16 @@ impl Bounds {
             _ => String::new(),
         }
     }
-    pub fn add_parameter(&mut self, name: &str, type_str: &str, bound_type: BoundType) -> bool {
-        if self.used.iter().any(|ref n| n.0 == name)  { return false; }
+    pub fn add_parameter(&mut self, name: &str, type_str: &str, bound_type: BoundType,
+                         is_nullable: bool) -> bool {
+        if self.used.iter().any(|ref n| n.0 == name) { return false; }
+        if bound_type == BoundType::Into {
+            if is_nullable == false { return true; }
```
Comment (EPashkin) [inline src/analysis/bounds.rs]: If `is_nullable` used only for this check, may be better move it to `type_for` or add separate check function that must called on `BoundType::Into`

Comment (GuillaumeGomez) [inline src/analysis/bounds.rs]: It seemed more logical this way. I still think we should keep in here to avoid having bounds logic splitted.

**Comment by EPashkin** [comment]
@GuillaumeGomez, cast `comments.into() as Option<&'a str>` really needed?
about_dialog.rs normally builds without it.

**Comment by GuillaumeGomez** [comment]
In a lot of cases it's needed. Don't know why unfortunately... Rust automatic coercion is a bit stupid. But at least, with this it works everytime.

**Comment by EPashkin** [comment]
I rebuild all modules without `as`. Affected `gio`, `glib`, `gtk`
And examples builds fine (after fix notebook.rs)
IMHO as `T: Into<Option<&'a str>>>(` function `T.info()` can only one result type.

**Comment by GuillaumeGomez** [comment]
In my case, it's gtk which didn't build without the explicit cast (I didn't try the other libs). I have the following rustc verison:

```
rustc 1.17.0-nightly (ea7a6486a 2017-02-04)
```

**Comment by EPashkin** [comment]
Strange my 2 versions builds fine. I add PR to gtk for CI test

rustc 1.17.0-nightly (0648517fa 2017-02-03)
binary: rustc
commit-hash: 0648517faf1e2cf37c8b6770cbd0180a816ed9a0
commit-date: 2017-02-03
host: x86_64-pc-windows-gnu
release: 1.17.0-nightly
LLVM version: 3.9

rustc 1.17.0-nightly (29dece1c8 2017-02-08)
binary: rustc
commit-hash: 29dece1c8bbebf7ae8034ef0826b119281730937
commit-date: 2017-02-08
host: x86_64-pc-windows-gnu
release: 1.17.0-nightly
LLVM version: 3.9

**Comment by GuillaumeGomez** [comment]
Ok, I'll try without the cast after updating the version locally. So strange...

**Comment by GuillaumeGomez** [comment]
After a little rustup update it works without the cast. Really strange. I remove the cast then!

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/analysis/bounds.rs [hunk 2]
```diff
@@ -44,14 +49,23 @@ impl Bounds {
             _ => None,
         }
     }
-    pub fn to_glib_extra(library: &Library, type_id: TypeId) -> String {
+    pub fn to_glib_extra(library: &Library, type_id: TypeId, is_nullable: bool) -> String {
```
Comment (EPashkin) [inline src/analysis/bounds.rs]: @GuillaumeGomez  While I prefer `library::Nullable` over `bool` I can merge as is.
You don't want squash or add Into for other options?

Comment (GuillaumeGomez) [inline src/analysis/bounds.rs]: > @GuillaumeGomez While I prefer library::Nullable over bool I can merge as is.

Then I'll update in this way.

> You don't want squash

I can, just tell me which commits you want me to keep.

> add Into for other options?

Not for now. :P

**Comment by EPashkin** [comment]
Thanks, I merge this.

**Comment by GuillaumeGomez** [comment]
Oh ok. I'll make a small cleanup then.

**Pull request merged**